### DISCUSSION TITLE
Correct package and package-type for .yaml files

### DISCUSSION THREAD

**Discussion by ackoch** [Netcentric/accesscontroltool#569] [Q&A] [answered]
What is the correct package and/or package type to place the .yaml files in? 

- I naively tried the "ui.config".. but this is a "container" that apparently does not allow non-osgi config files such as the yaml files [1] (and some filevault validator complains). 

- "ui.content" also is not correct, as [2] or [3] suggest the yaml files are supposed to be under /apps and ui.content is of type "content" in the latest aim archetype. Maybe "/conf" is a better location?  

- is "ui.apps" a good idea? I mean.. it's yaml files are configurations not code...

What is the best practice?

--

[1] https://jackrabbit.apache.org/filevault/packagetypes.html
[2] https://github.com/Netcentric/accesscontroltool/blob/develop/docs/Configuration.md
[3] https://www.netcentric.biz/insights/2020/01/ac-tool-with-aem-cloud-service.html

**Comment by kwin** [comment, marked as answer]
I would recommend putting it in an `application` package below `/apps/...`. Other configuration like OSGi configuration is placed below `/apps/...` as well. The only drawback is that install hooks are not allowed in `application` packages (https://issues.apache.org/jira/browse/JCRVLT-512). So you either need to disable the validation on that package or temporarily set it to type `mixed`. In any case the install hook is skipped for AEMaaCS as the installation there works fundamentally different: https://github.com/Netcentric/accesscontroltool/blob/develop/docs/ApplyConfig.md#installation-in-aem-as-a-cloud-service
